### DISCUSSION TITLE
Just curious: Does Play (Java) stops supporting ThreadLocal after 2.7.x?

### DISCUSSION THREAD

**Discussion by tanin47** [playframework/playframework#12994] [Q&A] [answered]
I've noticed that this particular part in 2.7.x:

```
Java code in Play uses a ThreadLocal to find out about contextual information such as the current HTTP request. Scala code doesn’t need to use ThreadLocals because it can use implicit parameters to pass context instead. ThreadLocals are used in Java so that Java code can access contextual information without needing to pass context parameters everywhere.
```

But it has been removed since 2.8.x.

Does this means PlayJava doesn't use ThreadLocal to find out about contextual information anymore? What does it use?

It seems this PR removes that part: https://github.com/playframework/playframework/commit/00eccef853d48bc669c6afd8ca45ffc73eda3f0a

**Comment by mkurz** [comment, marked as answer]
> Does this means PlayJava doesn't use ThreadLocal to find out about contextual information anymore? 

Exactly, since Play 2.8 ThreadLocal is not used anymore.

> What does it use?

Instead, the information that is needed needs to be passed on as parameters. (So what's new is that you can pass the current http request on to action methods).

For further explanation and examples please see: https://www.playframework.com/documentation/latest/JavaHttpContextMigration27

**Reply by tanin47** [reply]
Thank you so much. For context, I was integrating Togglz (didn't succeed) and encountered the part where I would need to use ThreadLocal. So, I was just wondering. This answer helps.
